diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,8 +3,7 @@ NULL =
 ACLOCAL_AMFLAGS = -I m4
 
 if ENABLE_SERVER
-    IPASERVER_SUBDIRS = ipaserver ipasphinx
-    SERVER_SUBDIRS = daemons init install
+    SERVER_SUBDIRS = daemons
 endif
 
 if WITH_IPATESTS
diff --git a/daemons/Makefile.am b/daemons/Makefile.am
--- a/daemons/Makefile.am
+++ b/daemons/Makefile.am
@@ -9,11 +9,7 @@ noinst_HEADERS = ipa-version.h.in
 
 SUBDIRS =			\
 	.			\
-	dnssec			\
-	ipa-kdb			\
-	ipa-slapi-plugins	\
 	ipa-sam			\
-	ipa-otpd		\
 	$(NULL)
 
 ipa-version.h: ipa-version.h.in $(top_builddir)/$(CONFIG_STATUS)
diff --git a/daemons/ipa-sam/ipa_sam.c b/daemons/ipa-sam/ipa_sam.c
--- a/daemons/ipa-sam/ipa_sam.c
+++ b/daemons/ipa-sam/ipa_sam.c
@@ -16,3 +16,3 @@
 
-#include <param.h>
+#include <sys/param.h>
 #include <ndr.h>
